glue_runner.py

```diff
@@ -0,0 +1,276 @@
+        "ExecutionProperty": {"MaxConcurrentRuns": 10},
+        "WorkerType": "Z.2X",
+        "NumberOfWorkers": glue_max_workers,
+        "Command": {
+            "Name": "glueray",
+            "Runtime": "Ray2.4",
+            "PythonVersion": "3.9",
+            "ScriptLocation": s3_glue_script_file_path,
+        },
+    }
+    region = args.region
+    glue_client = boto3.client(
+        "glue",
+        region,
+    )
+    try:
+        response = glue_client.get_job(JobName=glue_job_name)
+        print(f"Found Existing Glue Job `{glue_job_name}`: {response}")
+        print(f"Updating Existing Glue Job `{glue_job_name}`")
+        if not glue_iam_role:
+            glue_job_config["Role"] = response["Job"]["Role"]
+        glue_client.update_job(
+            JobName=glue_job_name,
+            JobUpdate=glue_job_config,
+        )
+    except ClientError as e:
+        print(f"Glue Job `{glue_job_name}` doesn't exist. Creating it.")
+        # create or update a glue job
+        glue_client.create_job(
+            Name=glue_job_name,
+            **glue_job_config,
+        )
+    print(f"New Glue Job Config: {glue_job_config}")
+
+    print(f"Getting Last Job Run Arguments for {glue_job_name}")
+    response = glue_client.get_job_runs(
+        JobName=glue_job_name,
+    )
+    job_runs = response["JobRuns"]
+    last_job_run_args = (
+        job_runs[0]["Arguments"] if job_runs and "Arguments" in job_runs[0] else {}
+    )
+    print(f"Last Job Run Args for {glue_job_name}: {last_job_run_args}\n\n")
+
+    cli_args = args.example_script_args
+    last_job_run_packages = last_job_run_args.get("--PACKAGE")
+    last_job_run_pip = last_job_run_args.get("--pip-install")
+    glue_start_job_run_args = {
+        "--PACKAGE": last_job_run_packages,
+        "--pip-install": last_job_run_pip,
+        "--STAGE": stage,
+        # set deltacat logging environment variables to be picked up by Glue
+        "--DELTACAT_APP_LOG_LEVEL": "DEBUG",
+        "--DELTACAT_SYS_LOG_LEVEL": "DEBUG",
+        "--DELTACAT_APP_LOG_DIR": "/home/glue-ray-engine/var/log/",
+        "--DELTACAT_SYS_LOG_DIR": "/home/glue-ray-engine/var/log/",
+        "--DELTACAT_APP_INFO_LOG_BASE_FILE_NAME": "customer_script.log.dc.app.info",
+        "--DELTACAT_SYS_INFO_LOG_BASE_FILE_NAME": "customer_script.log.dc.sys.info",
+        "--DELTACAT_APP_DEBUG_LOG_BASE_FILE_NAME": "customer_script.log.dc.app.debug",
+        "--DELTACAT_SYS_DEBUG_LOG_BASE_FILE_NAME": "customer_script.log.dc.sys.debug",
+    }
+
+    new_package = None
+    if dc_url:
+        new_package = f"deltacat @ {dc_url}"
+    elif dc_version:
+        new_package = f"deltacat=={dc_version}"
+    elif not (last_job_run_packages and last_job_run_pip):
+        # install the latest version of deltacat
+        new_package = "deltacat"
+    if new_package:
+        glue_start_job_run_args["--pip-install"] = new_package
+        glue_start_job_run_args["--PACKAGE"] = new_package
```

Comment:
Can you help me understand why we would need `--pip-install` as well as `--PACKAGE`. I believe the later is a Job parameter but curious to know do we try to install dependencies in the script again.

Reply:
The separate `--PACKAGE` arg is no longer needed, and has been removed (see related discussion with @chappidim above).